File: src/main/java/com/github/kdm1jkm/baseballSolver/condition/SingleCondition/BallCondition.java
package com.github.kdm1jkm.baseballSolver.condition.SingleCondition;

public class BallCondition implements SingleCondition {
    private final int num;
    private final int pos;

    public BallCondition(int num, int pos) {
        this.num = num;
        this.pos = pos;
    }

    public boolean isTrue(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == num && i != pos) return true;
        }
        return false;
    }

    public int getNum() {
        return num;
    }

    public int getPos(){
        return pos;
    }

    @Override
    public SingleCondition copy() {
        return new BallCondition(num, pos);
    }
}

File: src/main/java/com/github/kdm1jkm/baseballSolver/condition/singleCondition/BallCondition.kt
package com.github.kdm1jkm.baseballSolver.condition.singleCondition

import com.github.kdm1jkm.baseballSolver.condition.ConditionSort

class BallCondition(override val num: Int, override val pos: Int) : SingleCondition {
    override fun isTrue(nums: List<Int>): Boolean = nums[pos] != num && nums.any { it == num }

    override val sort: ConditionSort
        get() = ConditionSort.Ball
}